### PR TITLE
[ios] Create darwin/graphics to share graphics code between macos and ios

### PR DESCRIPTION
**Pull request by iskakaushik**
This change also moves both GL and Metal context and eternal texture components to `darwin/graphics`.

### CODE REVIEW

File: shell/platform/darwin/graphics/BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+assert(is_ios)
```
Comment (iskakaushik) [inline shell/platform/darwin/graphics/BUILD.gn:5]: I plan on making this folder be compatible with macOS as well. Metal code will be usable mostly as-is. I will wire the `shell_enable_metal` flag to also support mac.

File: shell/platform/darwin/graphics/ios_context.h [hunk 2]
```diff
@@ -12,7 +12,7 @@
 #include "flutter/fml/macros.h"
 #include "flutter/fml/platform/darwin/scoped_nsobject.h"
```
Comment (stuartmorgan) [inline shell/platform/darwin/graphics/ios_context.h:13]: The macOS code uses ARC; you can't share any `scoped_nsobject` headers with macOS at this point (making this work with ARC is non-trivial; I can find PR discussion later to point you to). I would strongly prefer we not put any scroped_nsobject code in a place where we intended to share with macOS.

**Review by stuartmorgan — COMMENTED** [review]
I haven't audited the implementation files here yet, but is it really the case that none of these are using engine internals that we are intentionally firewalling new embedder like desktop from? I see some reference to engine threads, for instance, which seems like a red flag.

**Comment by iskakaushik** [comment]
I will do this the other way around only moving the files we need for macOS metal.

**Pull request closed without merging**